metaci/build/autoscaling.py

```diff
@@ -114,8 +115,23 @@ def scale(self):
             resp = requests.patch(
                 url, json={"quantity": self.target_workers}, headers=headers
             )
+            if resp.json() and resp.json().get("id") == "cannot_update_above_limit":
+                limit = resp.json()["limit"]
+                resp = self.scale_max(url, headers, worker_type, limit)
+
             resp.raise_for_status()
```

Comment:
resp.raise_for_status() should happen before we try to parse the response as json.

Reply:
We don't want to throw the exception because we want to respond to the 422. Heroku error messages are JSON so it is safe to parse them.

But yeah, perhaps there is a bit more thought required here about if Heroku was literally down/unreachable or something and didn't return JSON: but the only improvement I can see would be around the clarity of the error message. Instead of "JSON failed to parse" we could return a bit more information about the response. And again, I think this is only in the case where there is a deep networking or infrastructure issue.

Reply:
Oh yeah I see. I think it's okay as is since the server formats errors as JSON too.